pkg/api/query/v1.go

```diff
@@ -427,25 +440,136 @@ func (qapi *QueryAPI) parseShardInfo(r *http.Request) (*storepb.ShardInfo, *api.
 	return &info, nil
 }
 
-func (qapi *QueryAPI) parseQueryExplainParam(r *http.Request, query promql.Query) (*engine.ExplainOutputNode, *api.ApiError) {
-	var explanation *engine.ExplainOutputNode
+func (qapi *QueryAPI) getQueryExplain(query promql.Query) (*engine.ExplainOutputNode, *api.ApiError) {
+	if eq, ok := query.(engine.ExplainableQuery); ok {
+		return eq.Explain(), nil
+	}
+	return nil, &api.ApiError{Typ: api.ErrorBadData, Err: errors.Errorf("Query not explainable")}
 
-	if val := r.FormValue(QueryExplainParam); val != "" {
-		var err error
-		enableExplanation, err := strconv.ParseBool(val)
-		if err != nil {
-			return explanation, &api.ApiError{Typ: api.ErrorBadData, Err: errors.Wrapf(err, "'%s' parameter", QueryExplainParam)}
+}
+
+func (qapi *QueryAPI) parseQueryAnalyzeParam(r *http.Request, query promql.Query) (queryTelemetry, error) {
+	if r.FormValue(QueryAnalyzeParam) != "" {
+		if eq, ok := query.(engine.ExplainableQuery); ok {
+			return processAnalysis(eq.Analyze()), nil
 		}
-		if enableExplanation {
-			if eq, ok := query.(engine.ExplainableQuery); ok {
-				explanation = eq.Explain()
-			} else {
-				return explanation, &api.ApiError{Typ: api.ErrorBadData, Err: errors.Errorf("Query not explainable")}
-			}
+	}
+	return queryTelemetry{}, errors.Errorf("Query not analyzable; change engine to 'thanos' ")
+}
```

Comment:
Seems like we are always including an `analyze` param via UI, so this no longer works and always errors. So let's change the logic a bit.
```suggestion
func (qapi *QueryAPI) parseQueryAnalyzeParam(r *http.Request, query promql.Query) (queryTelemetry, error) {
	if r.FormValue(QueryAnalyzeParam) == "true" || r.FormValue(QueryAnalyzeParam) == "1" {
		if eq, ok := query.(engine.ExplainableQuery); ok {
			return processAnalysis(eq.Analyze()), nil
		}

		return queryTelemetry{}, errors.Errorf("Query not analyzable; change engine to 'thanos'")
	}

	return queryTelemetry{}, nil
}
```